src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -410,18 +410,18 @@ private static unsafe nuint GetIndexOfFirstNonAsciiByte_Sse2(byte* pBuffer, nuin
 
             if ((bufferLength & 8) != 0)
             {
-                if (Bmi1.X64.IsSupported)
+                if (UIntPtr.Size == sizeof(ulong))
```

Comment:
Again, this path should be faster on all supported 64-bit platforms, but I wasn't sure how that's best expressed here.  Should this use conditional compilation, or an explicit `if (X86Base.X64.IsSupported || ArmBase.Arm64.IsSupported)`?